src/main/scala/org/broadinstitute/hail/expr/Type.scala

```diff
@@ -223,7 +224,8 @@ case object TDouble extends TNumeric {
   override def genValue: Gen[Annotation] = arbitrary[Double]
 
   override def valuesSimilar(a1: Annotation, a2: Annotation): Boolean =
-    a1 == a2 || (a1 != null && a2 != null && D_==(a1.asInstanceOf[Double], a2.asInstanceOf[Double]))
+  // a1 == a2 || (a1 != null && a2 != null && D_==(a1.asInstanceOf[Double], a2.asInstanceOf[Double]))
+    a1 == a2 || (a1 != null && a2 != null && str(a1) == str(a2))
```

Comment:
These two changes seem odd, I don't understand why we're converting to a string. In the case we really want this I don't think we need the null checks since null gets converted to `"NA"`.